### PR TITLE
Implements getModelVersion in the java client

### LINKED ISSUE #6603: [FR] Implement `getModelVersion` in Java client
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

Implement `getModelVersion` in Java client to provide users a way to get a model version in Java.

#### References

- [`get_model_version` in Python](https://www.mlflow.org/docs/latest/python_api/mlflow.client.html#mlflow.client.MlflowClient.get_model_version)

### Motivation

> #### What is the use case for this feature?

To provide users a way users to get a model version in Java.

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### Details

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [X] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by wgottschalk**
## Related Issues/PRs

Close https://github.com/mlflow/mlflow/issues/6603

## What changes are proposed in this pull request?

Adds support for the getModelVersion api but exposing it in the client and adding parsing support in the protobuf mapper.

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [x] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@wgottschalk Thanks for the PR! Can you add a test?

**Comment by wgottschalk** [comment]
No problem! I'll be able to get to this next weekend

**Comment by dbczumar** [comment]
> No problem! I'll be able to get to this next weekend

Hi @wgottschalk, just following up here. Is it possible to add a test?

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java [hunk 1]
```diff
@@ -160,8 +160,16 @@ String makeCreateModelVersion(String modelName, String runId, String source) {
     return print(builder);
   }
 
-  String makeGetModelVersionDetails(String modelName, String version) {
-    return print(GetModelVersion.newBuilder().setName(modelName).setVersion(version));
```
Comment (wgottschalk) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java:164]: good thing tests were added, turns out this is syntax is used for post requests and get requests

File: mlflow/java/client/src/test/java/org/mlflow/tracking/ModelRegistryMlflowClientTest.java [hunk 2]
```diff
@@ -13,18 +23,6 @@
 import org.testng.annotations.BeforeTest;
 import org.testng.annotations.Test;
 
-import java.io.File;
```
Comment (wgottschalk) [inline mlflow/java/client/src/test/java/org/mlflow/tracking/ModelRegistryMlflowClientTest.java:16]: intellij formatted this part of the file. Is there something like spotless that I should be running on these files to keep the import ordering correct?

Comment (dbczumar) [reply mlflow/java/client/src/test/java/org/mlflow/tracking/ModelRegistryMlflowClientTest.java:16]: This is just fine with me :)

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @wgottschalk ! Before merging, can you sign your commits (https://github.com/mlflow/mlflow/pull/6955/checks?check_run_id=8925139196)?

**Comment by wgottschalk** [comment]
@dbczumar commits should be signed off now!

**Comment by dbczumar** [comment]
Thank you, @wgottschalk ! Apologies for the delay :)

**Pull request merged**